fdt.cc

```diff
@@ -1414,13 +1423,38 @@ device_tree::resolve_cross_references(uint32_t &phandle)
 				return;
 			}
 		}
+		// Track referenced nodes for later garbage collection
+		referenced_nodes.push_back(target);
```

Comment:
I don't think this is quite right, consider two `/delete-if-unreferenced/` nodes that refer to each other.  They won't be GC'd, even though nothing else references them.  You probably need to do a separate tree walk and skip nodes with the `delete_if_unreferenced` property.